src/components/Tool/FixPhoneNumbers/helper.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  PersonPhoneNumberInput,
+  PersonUpdateInput,
+} from 'src/graphql/types.generated';
+import { FormValuesPerson } from './FixPhoneNumbers';
+
+export const determineBulkDataToSend = (
+  values: FormValuesPerson[],
+  defaultSource: string,
+): PersonUpdateInput[] => {
+  const dataToSend = [] as PersonUpdateInput[];
+
+  values.forEach((value) => {
+    const primaryNumber = value.phoneNumbers.nodes.find(
+      (number) => number.source === defaultSource,
+    );
+    if (primaryNumber) {
+      dataToSend.push({
+        id: value.id,
+        phoneNumbers: value.phoneNumbers.nodes.map(
+          (number) =>
+            ({
+              id: number.id,
+              primary: number.id === primaryNumber.id,
+              number: number.number,
+              validValues: true,
+            } as PersonPhoneNumberInput),
```

Comment:
This `as` type cast tells TypeScript to ignore whatever properties this object has and just treat it like a `PersonPhoneNumberInput`, which I don't think is what we want. I believe you can just remove it.